Assets/Mirror/Runtime/NetworkConnection.cs

```diff
@@ -32,6 +32,11 @@ public class NetworkConnection : IDisposable
         /// </remarks>
         public int connectionId = -1;
 
+        /// <summary>
+        /// Flag that indicates the connection has been authenticated
+        /// </summary>
+        public bool isAuthenticated;
```

Comment:
LOL,  I was so over-engineering this one flag in my head.  This is nice and simple